tests/components/zwave_js/test_binary_sensor.py

```diff
@@ -84,3 +87,76 @@ async def test_notification_sensor(hass, multisensor_6, integration):
     assert state
     assert state.state == STATE_ON
     assert state.attributes["device_class"] == DEVICE_CLASS_MOTION
+
+
+async def test_property_sensor_door_status(hass, lock_august_pro, integration):
+    """Test property binary sensor with sensor mapping (doorStatus)."""
+    node = lock_august_pro
+
+    state = hass.states.get(PROPERTY_DOOR_STATUS_BINARY_SENSOR)
+    assert state is not None
+    assert state.state == STATE_OFF
+
+    # open door
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 6,
+            "args": {
+                "commandClassName": "Door Lock",
+                "commandClass": 98,
+                "endpoint": 0,
+                "property": "doorStatus",
+                "newValue": "open",
+                "prevValue": "closed",
+                "propertyName": "doorStatus",
+            },
+        },
+    )
+    node.receive_event(event)
+    state = hass.states.get(PROPERTY_DOOR_STATUS_BINARY_SENSOR)
+    assert state.state == STATE_ON
+
+    # close door
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 6,
+            "args": {
+                "commandClassName": "Door Lock",
+                "commandClass": 98,
+                "endpoint": 0,
+                "property": "doorStatus",
+                "newValue": "closed",
+                "prevValue": "open",
+                "propertyName": "doorStatus",
+            },
+        },
+    )
+    node.receive_event(event)
+    state = hass.states.get(PROPERTY_DOOR_STATUS_BINARY_SENSOR)
+    assert state.state == STATE_OFF
+
+
+async def test_property_sensor_no_sensor_mapping(
+    hass, client, lock_august_pro, integration
+):
+    """Test property sensor without sensor mapping."""
+    node = lock_august_pro
+
+    info = ZwaveDiscoveryInfo(
+        node=node,
+        primary_value=node.values["6-98-00-outsideHandlesCanOpenDoorConfiguration-00"],
+        platform="binary_sensor",
+        platform_hint="property",
+    )
+    zwave_property_binary_sensor = ZWavePropertyBinarySensor(
```

Comment:
We shouldn't interact with the platform details directly in the tests. Please use the core interfaces to set up and start the test. We need a fixture without supported sensor mapping and then assert that it results in a disabled entity.

Reply:
@MartinHjelmare i was trying to think through a test like that but I couldn’t think of a way of doing it because then would need to add a ZWaveDiscoverySchema in discovery that doesn’t have a corisponding sensor mapping in binary sensor. And since I couldn’t think of a real world example to add a ZWaveDiscoverySchema for I did this. I thought about removing the lines in question, but then you have indeterminate behavior when/if there ever is a discovery scheme that doesn’t have a coriaponding sensor mapping. This isn’t a problem for the notification binary sensor due to the multiplicity of the command class notification values amd how those mappings are mapped. But for the property binary sensor, the property name either matches or it doesn’t. Or maybe I’m missing something obvious. If you have some idea how to test the case differently suggestions please cuz I racked my brain for a while

Reply:
@MartinHjelmare actuallly....i may have thought of a way to change the code to make the test not necessary since the code path *should* be imposible anyway

Reply:
@MartinHjelmare that should do it. refactor makes it so there is no untested code paths and we dont need to invent a scenario where there is a `ZWaveDiscoverySchema` in `discovery.py` with `platform` of `binary_sensor` and `hint` of `property` without a corresponding `PropertySensorMapping` in `binary_sensory.py`